### ISSUE TITLE
lock continues to attempt even after fail timeout

### ISSUE THREAD

**Issue by chris-digitalrose** [AmrSaber/simple-redis-mutex#6]
Issue is when the failAfterMillis option is used to timeout a lock attempt,  the module properly rejects the promise with the time out error.  However the retry attempt timeout continues to fire so the library keeps trying to set the lock in redis, even though there is no client waiting on it anymore.   This problem ended up putting orphan locks in redis.

Hacked around the problem with a manual fork by saving an attemptTimeoutId and clearing the timeout before the reject.

```
    // Set time out to fail acquiring the lock if it's sent
    if (failAfterMillis != null) {
      failTimeoutId = setTimeout(
        () => { 
          if (attemptTimeoutId != null) { clearTimeout(attemptTimeoutId); }   // ADDED
          reject(new Error(`Lock could not be acquire for ${failAfterMillis} millis`)); 
        }, failAfterMillis);
    }
```

Thanks
Chris

**Comment by AmrSaber** [comment]
You are welcome to make a PR with that fix. But if so, please add a test case for this case. Also, please make the PR to dev.

**Comment by AmrSaber** [comment]
@chris-digitalrose Any intention of working on a PR, or should I start working on it?

**Comment by chris-digitalrose** [comment]
Hi Amr,  Please go ahead.  I'm new to github and PR's and automated testing

**Comment by AmrSaber** [comment]
@chris-digitalrose The bug should be fixed now in the latest release (v 1.0.3) and I updated existing tests to include the case you mentioned. Thank you very much for pointing out the bug.

**Comment by chris-digitalrose** [comment]
Thank you for the fix @AmrSaber,  we'll slide this new rev into our next update.
